sklearn/utils/tests/test_extmath.py

```diff
@@ -454,24 +454,33 @@ def naive_log_logistic(x):
     assert_array_almost_equal(log_logistic(extreme_x), [-100, 0])
 
 
-def test_incremental_weighted_mean_and_variance_simple():
-    rng = np.random.RandomState(42)
+@pytest.fixture()
+def rng():
+    return np.random.RandomState(42)
+
+
+def test_incremental_weighted_mean_and_variance_simple(rng):
     mult = 10
     X = rng.rand(1000, 20)*mult
     sample_weight = rng.rand(X.shape[0]) * mult
     mean, var, _ = _incremental_weighted_mean_and_var(X, sample_weight,
                                                       0, 0, 0)
 
-    mean_exp = np.average(X, weights=sample_weight, axis=0)
-    var_exp = np.average(X**2, weights=sample_weight, axis=0) - mean_exp**2
-    assert_almost_equal(mean, mean_exp)
-    assert_almost_equal(var, var_exp)
+    expected_mean = np.average(X, weights=sample_weight, axis=0)
+    expected_var = np.average(X**2, weights=sample_weight, axis=0) - \
+        expected_mean**2
+    assert_almost_equal(mean, expected_mean)
+    assert_almost_equal(var, expected_var)
 
 
-def test_incremental_weighted_mean_and_variance():
+@pytest.mark.parametrize("mean", [0, 1e7, -1e7])
+@pytest.mark.parametrize("var", [1, 1e-8, 1e5])
+@pytest.mark.parametrize("loc, scale", [
```

Comment:
maybe name these as weight_loc and weight_scale to not be confused with `mean` and `var`.

Reply:
@jnothman thanks for your review. This is done.